### PR TITLE
add pandas_utils - extract traces from mcmc to dataframe

### PR DESCRIPTION
**Pull request by sashaostr**
add util to create dataframe from mcmc traces

### CODE REVIEW

**Comment by fonnesbeck** [comment]
Thanks for the PR. Does this work win multidimensional arrays? This utility should be accompanied by relevant unit tests.

**Comment by sashaostr** [comment]
Hi,
I added support for md arrays and unitests.
Just one problem that I wasn't able to run the tests from the project - I'm not familiar with numpy tests 
and wasn't able to find how to run it without spending hours...So if you can point me how to do it I"ll able to test the project. But actually tests are passing when I ran manually.
 Thanks!

**Comment by fonnesbeck** [comment]
Is this still ongoing work? If so, is it ready?

**Comment by sashaostr** [comment]
yes, it is ready.

**Comment by sashaostr** [comment]
mistakenly  closed and reopened..
The work is ready.

File: pymc/pandas_utils.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
+'''
+Created on Dec 16, 2015
+
+@author: alexander ostrikov
+'''
+
+import pandas as pd
+import numpy as np
+
+def mcmc_traces_to_dataframe(mcmc):
```
Comment (fonnesbeck) [inline pymc/pandas_utils.py:10]: This is a rather long, awkward name. Something shorter perhaps? e.g `traces_to_df`

File: pymc/pandas_utils.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    def get_data(trace):
+        shape = trace.shape
+        if len(shape) == 1:
+            return trace
+        elif len(shape) == 2 and shape[1] == 1:
+            return trace[:, 0]
+        elif len(shape) == 2:
```
Comment (fonnesbeck) [inline pymc/pandas_utils.py:17]: You have already checked for `len(shape)` above. Better to handle this there.

**Comment by fonnesbeck** [comment]
It might be better to implement this as a `MCMC` method, especially since it takes a `MCMC` argument. Then you would be able to:

```
M = MCMC(<model>)
M.sample(n)
df = M.traces_to_df()
```

The `Sampler` class already has a `write_csv` method; this would work analogously.

**Comment by fonnesbeck** [comment]
Closed due to inactivity. Please reopen if interested.

File: pymc/pandas_utils.py (hunk 1 above)
Comment (sashaostr) [reply pymc/pandas_utils.py:10]: sure I"ll change

File: pymc/pandas_utils.py (hunk 2 above)
Comment (sashaostr) [reply pymc/pandas_utils.py:17]: do you mean putting len(shape) into variable and then reuse in if's?

**Comment by fonnesbeck** [comment]
@sashaostr do you want me to reopen this?

File: pymc/pandas_utils.py (hunk 2 above)
Comment (fonnesbeck) [reply pymc/pandas_utils.py:17]: Use a nested conditional, so you don't have to check the same condition again:

```
elif len(shape)==2:
    if shape[1]==1:
        return trace[:, 0]
    return trace.tolist()
```

**Pull request closed without merging**